hera_qm/xrfi.py

```diff
@@ -1639,6 +1658,310 @@ def xrfi_run(ocalfits_file, acalfits_file, model_file, data_file, history,
+
+    Parameters
+    ----------
+    ocalfits_files : str
+        The omnical calfits files to use to flag on gains and chisquared values.
+    acalfits_files : str
+        The abscal calfits files to use to flag on gains and chisquared values.
+    model_files : str
+        THe model visibility files to flag on.
+    data_files : str
+        The raw visibility data files to flag.
+    history : str
+        The history string to include in files.
+    xrfi_path : str, optional
+        Path to save xrfi files to. Default is a subdirectory "{JD}/" inside
+        the same directory as data_file.
+    kt_size : int, optional
+        The size of kernel in time dimension for detrend in xrfi algorithm.
+        Default is 8.
+    kf_size : int, optional
+        Size of kernel in frequency dimension for detrend in xrfi algorithm.
+        Default is 8.
+    sig_init : float, optional
+        The starting number of sigmas to flag on. Default is 5.0.
+    sig_adj : float, optional
+        The number of sigmas to flag on for data adjacent to a flag.
+        Default is 2.0.
+    ex_ants : str, optional
+        A comma-separated list of antennas to exclude. Flags of visibilities formed
+        with these antennas will be set to True. Default is None (i.e., no antennas
+        will be excluded).
+    metrics_file : str, optional
+        Metrics file that contains a list of excluded antennas. Flags of visibilities
+        formed with these antennas will be set to True. Default is None (i.e.,
+        no antennas will be excluded).
+    clobber : bool, optional
+        If True, overwrite existing files. Default is False.
+    run_check : bool
+        Option to check for the existence and proper shapes of parameters
+        on UVFlag Object.
+    check_extra : bool
+        Option to check optional parameters as well as required ones.
+    run_check_acceptability : bool
+        Option to check acceptable range of the values of parameters
+        on UVFlag Object.
+
+    Returns
+    -------
+    None
+
+    """
+    history = 'Flagging command: "' + history + '", Using ' + hera_qm_version_str
+    xants = process_ex_ants(ex_ants=ex_ants, metrics_file=metrics_file)
+
+    # Make sure input files are sorted
+    ocalfits_files = sorted(ocalfits_files)
+    acalfits_files = sorted(acalfits_files)
+    model_files = sorted(model_files)
+    data_files = sorted(data_files)
+
+    # Initial run on cal data products
+    # Calculate metric on abscal data
+    uvc_a = UVCal()
+    uvc_a.read_calfits(acalfits_files)
+    uvf_apriori = UVFlag(uvc_a, mode='flag', copy_flags=True, label='A priori flags.')
+    uvf_ag, uvf_agf = xrfi_pipe(uvc_a, alg='detrend_medfilt', Kt=kt_size, Kf=kf_size, xants=xants,
+                                cal_mode='gain', sig_init=sig_init, sig_adj=sig_adj,
+                                label='Abscal gains, round 1.', pad=False,
+                                run_check=run_check,
```

Comment:
If you define checks = {'run_check': run_check} etc. than you can just ** these kwargs in and save some lines.

Reply:
Or go broader: make every argument that never changes get passed via a single dict.